### ISSUE TITLE
MixNet Paper Does Not Mention 1x1 Grouped Convolutions

### ISSUE THREAD

**Issue by andravin** [tensorflow/tpu#513]
https://github.com/tensorflow/tpu/blob/bb4caadc0ec7ee88e1cb1d560999ddbc6958966a/models/official/mnasnet/mixnet/mixnet_builder.py#L142-L143

Reading the above source code for MixNet, I was surprised to see that many of the expansion and projection 1x1 convolution layers use 2 convolution groups. This is not mentioned in the paper.

The text `a1.1` indicates that the expansion layer uses 2 convolution groups, each with filter size 1. Likewise, `p1.1` indicates that the projection layer uses 2 convolution group with filter size 1.

Why does the paper not mention the grouped convolutions?

**Comment by mingxingtan** [comment]
Good point. I did mention our search settings are similar to MnasNet (not using group conv) and FBNet (using group conv), but I might forgot to make it explicit (Will update the paper to make it explicit)

Notably, our group conv is very simple (only have 1 or 2 groups with no channel shuffling).

**Comment by andravin** [comment]
It made sense to try group convolutions of course, as many architectures have shown them to improve model efficiency. I did notice that the MixNet reference implementation uses just 1 or 2 groups, same as FBNet.

But this begs the question, if the FBNet paper shows us that group 1x1 convolutions are a known enhancement to MobileNetV2, then how much of the MixNet advantage versus MobileNeV2/MNasnet is actually due to mixed convolutions, and how much is due to grouped 1x1? 

Likewise, FBNet does not use squeeze-and-excitation or swish activation, so how much of the MixNet advantage versus FBNet is due to mixed convolution, and how much is due to these other optimizations?

**Comment by mingxingtan** [comment]
You brought up a good point: in order to understand the impact of each factor, it needs to run multiple architecture search experiments, which is relatively expensive. In addition to that, a strict comparison would require exactly the same search algorithm and the same search space, which is non-trivial. 

However, as an indirect evidence, we have shown the improvement of mixconv on MobileNetV1 and MobileNetV2 as well (exactly the same training settings and networks except DepthwiseConv vs MixConv), which is more like a fair comparison.

Sorry, I guess I still couldn't give you the answer you want unless adding more experiments. I will try that when I have the bandwidth; I also encourage you trying different settings and sharing us your findings (if possible).

**Comment by andravin** [comment]
Well, I think maybe I got ahead of the issue a bit. Perhaps the main thing is just to document which layers use group 1x1 convolutions in MixNet-S,M,L so that the paper contains a complete definition of the networks.

I see how experiments become more expensive when architecture search is involved. Maybe there is a simple ablation study that could be done with MobileNetV2 + 1x1 group convolution + mixed convolution, similar to the what you already did with MobileNetV2 + mixed convolution.

I do not have the resources to run these experiments at the moment. I appreciate how time consuming experiments are, even with the substantial resources at your disposal. 

Anyway, thank you for being so responsive. I really enjoyed the paper, and now that I understand it uses group convolutions, I see connections with my own research that were not apparent before. Also, your ability to keep designs simple is an ideal to which we all should aspire.

**Comment by mingxingtan** [comment]
I will definitely update the paper to spell 1x1 conv out. Adding one more experiment with 1x1 group conv + MobileNetV2 seems reasonable. Will give it a try soon.

**Comment by andravin** [comment]
Would also be good to know the variant of MobileNetV2 that was used for the ablation studies. It appears to be different from baseline MobileNetV2 (72% at 300 MFLOPS in Table 2), because the accuracy is much higher (~72.9% at ~310 MFLOPS, in Figires 1.b, 4.b., 6.b.).

**Comment by orashi** [comment]
It would be really kind if it's possible for you to referrence & mention the usage of swish activation & SE module in the paper. As shwon in [AtomNAS](https://openreview.net/forum?id=BylQSxHFwr) Table 1, these tricks are likely to contribute a considerable 1.0%+ Top1 acc boost in the imagenet task.

**Comment by andravin** [comment]
Just noticed that the paper was updated December 1st to mention the details of the network architecture search space:

>Our search space also includes swish activation [4, 20], squeeze-and-excitation module [8] (similar to [29]), and grouped convolutions with group size 1 or 2 for 1x1 convolutions (similar to [30]).
https://arxiv.org/abs/1907.09595

I don't see additional experiments or detailed network architectures (the architecture diagrams still do not show which layers use group 1x1 convolutions, for example).

Nevertheless, I am closing this issue because the main point was addressed, and we can refer to the source code for the complete details.

Thanks!
